lib/errors/oauth-error.js

```diff
@@ -9,33 +9,33 @@ const http = require('http');
  * Constructor.
  */
 
-function OAuthError(messageOrError, properties) {
-  let message = messageOrError instanceof Error ? messageOrError.message : messageOrError;
-  const error = messageOrError instanceof Error ? messageOrError : null;
-  if (properties == null || !Object.entries(properties).length ) {
-    properties = {};
-  }
+class OAuthError extends Error {
+	constructor(messageOrError, properties) {
+		let message = messageOrError instanceof Error ? messageOrError.message : messageOrError;
+		const error = messageOrError instanceof Error ? messageOrError : null;
+		if (properties == null || !Object.entries(properties).length) {
```

Comment:
no need for this if clause just use spread operator. if properties is null, it would still generate the correct Object